tests/ignite/metrics/test_ndcg.py

```diff
@@ -0,0 +1,145 @@
+import numpy as np
+import pytest
+import torch
+from sklearn.metrics import ndcg_score
+from sklearn.metrics._ranking import _dcg_sample_scores
+
+from ignite.exceptions import NotComputableError
+from ignite.metrics.recsys.ndcg import NDCG
+
+
+@pytest.fixture(params=[item for item in range(6)])
+def test_case(request):
+
+    return [
+        (torch.tensor([[3.7, 4.8, 3.9, 4.3, 4.9]]), torch.tensor([[2.9, 5.6, 3.8, 7.9, 6.2]])),
+        (
+            torch.tensor([[3.7, 3.7, 3.7, 3.7, 3.7], [3.7, 3.7, 3.7, 3.7, 3.9]]),
+            torch.tensor([[1.0, 2.0, 3.0, 4.0, 5.0], [1.0, 2.0, 3.0, 4.0, 5.0]]),
+        ),
+    ][request.param % 2]
+
+
+@pytest.mark.parametrize("k", [None, 2, 3])
+@pytest.mark.parametrize("exponential", [True, False])
+@pytest.mark.parametrize("ignore_ties_flag", [True, False])
+@pytest.mark.parametrize("ignore_ties_input", [True, False])
+def test_output_cpu(test_case, k, exponential, ignore_ties_flag, ignore_ties_input):
+
+    device = "cpu"
+    y_pred_distribution, y_true = test_case
+
+    y_pred = torch.multinomial(y_pred_distribution, 5, replacement=not ignore_ties_input)
+
+    if not ignore_ties_input and ignore_ties_flag:
+        return
+
+    ndcg = NDCG(k=k, device=device, exponential=exponential, ignore_ties=ignore_ties_flag)
+    ndcg.update([y_pred, y_true])
+    result_ignite = ndcg.compute()
+
+    if exponential:
+        y_true = 2 ** y_true - 1
+
+    result_sklearn = ndcg_score(y_true.numpy(), y_pred.numpy(), k=k, ignore_ties=ignore_ties_flag)
+
+    np.testing.assert_allclose(np.array(result_ignite), result_sklearn, rtol=2e-6)
+
+
+@pytest.mark.parametrize("k", [None, 2, 3])
+@pytest.mark.parametrize("exponential", [True, False])
+@pytest.mark.parametrize("ignore_ties_flag", [True, False])
+@pytest.mark.parametrize("ignore_ties_input", [True, False])
+@pytest.mark.skipif(torch.cuda.device_count() < 1, reason="Skip if no GPU")
+def test_output_gpu(test_case, k, exponential, ignore_ties_flag, ignore_ties_input):
```

Comment:
Let's rename this test : test_output_gpu -> test_output_on_cuda such that it would run on GPU: https://github.com/pytorch/ignite/blob/master/CONTRIBUTING.md#codebase-structure